### PR TITLE
Remove Environment.resolveConfig

### PR DESCRIPTION
**Pull request by rjernst**
This method has multiple modes of resolving config files by
first looking in the config directory, then on the classpath,
and finally by prefixing with "config/" on the classpath.

Most of the places taking advantage of this were tests, so they
did not have to setup a real home dir with config. The only place
that was really relying on it was the code which loads names.txt
to randomly choose a node name.

This change fixes test to setup fake home dirs with their config
files. It also makes the logic for finding names.txt explicit:
look in config dir, and if it doesn't exist, load /config/names.txt
from the classpath.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/Names.java [hunk 1]
```diff
@@ -35,23 +38,20 @@
...
                 }
             }
-            try (BufferedReader reader = FileSystemUtils.newBufferedReader(nodeNames, Charsets.UTF_8)) {
-                int number = ((ThreadLocalRandom.current().nextInt(numberOfNames)) % numberOfNames);
-                for (int i = 0; i < number; i++) {
-                    reader.readLine();
-                }
-                return reader.readLine();
-            }
+            int index = ((ThreadLocalRandom.current().nextInt(names.size())) % names.size());
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/common/Names.java]: I know you did not add it, but taking the remainder seems unnecessary?

**Comment by jpountz** [comment]
LGTM

**Comment by s1monw** [comment]
LGTM this really should go into beta2

**Comment by nik9000** [comment]
LGTM

**Comment by rjernst** [comment]
Pushed to 2.0 branch as well: bf0fe57

**Pull request merged**